Framework/PythonInterface/core/src/PythonStdoutChannel.cpp

```diff
@@ -14,28 +14,30 @@
 #include <boost/iostreams/stream.hpp>
 
 namespace { // anonymous namespace
+
 class PyStdoutSink {
 public:
   typedef char char_type;
   typedef boost::iostreams::sink_tag category;
 
   std::streamsize write(const char *s, std::streamsize n) {
+    Mantid::PythonInterface::GlobalInterpreterLock gil; // acquire the GIL
     // PySys_WriteStdout truncates to 1000 chars
     static const std::streamsize MAXSIZE = 1000;
 
     std::streamsize written = std::min(n, MAXSIZE);
     PySys_WriteStdout((boost::format("%%.%1%s") % written).str().c_str(), s);
 
     return written;
-  }
+  } // release the GIL
 };
 
 // wrapper of that sink to be a stream
 PyStdoutSink pyStdoutSinkInstance = PyStdoutSink(); // needs to be initialized separately
-boost::iostreams::stream<PyStdoutSink> PyStdout(pyStdoutSinkInstance);
+boost::iostreams::stream<PyStdoutSink> PyStdoutStream(pyStdoutSinkInstance);
 
 } // anonymous namespace
 
 namespace Poco {
-PythonStdoutChannel::PythonStdoutChannel() : ConsoleChannel(PyStdout) {}
+PythonStdoutChannel::PythonStdoutChannel() : ConsoleChannel(PyStdoutStream) {}
```

Comment:
This sounds a "fix" to a bug.

Reply:
I just renamed the variable to explicitly state that this object *is* an object of type `std::stream`